tests/testsuite/update.rs

```diff
@@ -1646,21 +1646,124 @@ fn update_breaking_unstable() {
         .file(
             "Cargo.toml",
             r#"
+            [package]
+            name  =  "foo"
+            version  =  "0.0.1"
+            edition  =  "2015"
+            authors  =  []
+        "#,
+        )
+        .file("src/lib.rs", "")
+        .build();
+
+    p.cargo("update --breaking")
+        .masquerade_as_nightly_cargo(&["update-breaking"])
+        .with_status(101)
+        .with_stderr_unordered(
+            "\
+[ERROR] the `--breaking` flag is unstable, pass `-Z unstable-options` to enable it
+See https://github.com/rust-lang/cargo/issues/12425 for more information about the `--breaking` flag.
+",
+        )
+        .run();
+}
+
+#[cargo_test]
+fn update_breaking_dry_run() {
+    Package::new("incompatible", "1.0.0").publish();
+    Package::new("ws", "1.0.0").publish();
+
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+                # Check if formatting is preserved. Nothing here should change, due to dry-run.
+
+                [workspace]
+                members  =  ["foo"]
+
+                [workspace.dependencies]
+                ws  =  "1.0"  # Preserve formatting
+            "#,
+        )
+        .file(
+            "foo/Cargo.toml",
+            r#"
+                # Check if formatting is preserved. Nothing here should change, due to dry-run.
+
                 [package]
                 name  =  "foo"
                 version  =  "0.0.1"
                 edition  =  "2015"
                 authors  =  []
+
+                [dependencies]
+                incompatible  =  "1.0"  # Preserve formatting
+                ws.workspace  =  true  # Preserve formatting
             "#,
         )
-        .file("src/lib.rs", "")
+        .file("foo/src/lib.rs", "")
         .build();
 
-    p.cargo("update --breaking")
+    p.cargo("generate-lockfile").run();
+
+    Package::new("incompatible", "1.0.1").publish();
+    Package::new("ws", "1.0.1").publish();
+
+    Package::new("incompatible", "2.0.0").publish();
+    Package::new("ws", "2.0.0").publish();
+
+    p.cargo("update -Zunstable-options --dry-run --breaking")
         .masquerade_as_nightly_cargo(&["update-breaking"])
-        .with_status(1)
-        .with_stderr_contains("[ERROR] unexpected argument '--breaking' found")
+        .with_stderr(
+            "\
+[UPDATING] `dummy-registry` index
+[UPGRADING] incompatible ^1.0 -> v2.0.0
+[UPGRADING] ws ^1.0 -> v2.0.0
+[LOCKING] 2 packages to latest compatible versions
+[UPDATING] incompatible v1.0.0 -> v2.0.0
+[UPDATING] ws v1.0.0 -> v2.0.0
```

Comment:
Do we need both UPGRADING and UPDATING?  Is UPDATING enough to imply UPGRADING?

They are covering distinct aspects so I'm not entirely sure on this.

Reply:
I have changed the upgrade message to show the updated version requirement rather than the latest version, so the upgrade and update messages are more distinct now. But I am fine with dropping one or the other if you like.

```
[UPGRADING] incompatible ^1.0 -> ^2.0
[UPDATING] incompatible v1.0.0 -> v2.0.0
```

Reply:
Having the messages more distinct helps.  Part of the challenge (and why we didn't want a separate `cargo upgrade` command) is the distinction between an "update" and an "upgrade" can be confusing to users.  This also causes a lot more output.  However, people also want to know when things are happening.

How about we keep this for now and use the post-merge call-for-testing to see what feedback we get.  If we don't have it, the conversation is only about adding it or not.  If we have it, the conversation is about whether we should have it and how we can tweak what we have.